nilearn/decoding/decoder.py

```diff
@@ -1338,6 +1347,10 @@ def __init__(
             n_jobs=n_jobs,
         )
 
+    def score(self, X, y, *args):
+        """Call score method from base class."""
+        return _BaseDecoder.score(self, X, y, *args)
```

Comment:
OK I see. In terms of MRO, it means that we favor the `score` define in the mixin instead of the base class. Normally, this is usually what we want. But this is not the case, here. However, if we put the mixin on the right, we will break all the tag system.

Since `_BaseDecoder` is defining the score function, it means that you actually don't want to have the functionality defined by the mixin so you don't need to inherate. So from this point, the right fix to me is the following:

```python
class _BaseDecoder(...):
    def score(self, ...):
        ...
    def _more_tags(self):
        return {"require_y": True}

class Decoder(_BaseDecoder):
    _estimator_type = "classifier"

    ...

class DecoderRegressor(MultiOutputMixin, _BaseDecoder):
    _estimator_type = "regressor"

    ...
```

The use of the `RegressorMixin` and `ClassifierMixin` in your case is just required to defined the `_estimator_type` so let's define it directly in the children class. The second thing that we want is the definition of the `"require_y"` tag that can be done in the base class since this is common to both mixins.

I did not foresee the part or redefining the `score` function. Let me know if this is making sense to you @ymzayek.

Reply:
> I did not foresee the part or redefining the score function. Let me know if this is making sense to you @ymzayek.

I did not either and then I tried to go around it with redefining `score` but you are right, we do not need those mixins and can just define the needed attribute and tag function as you suggest. I was coming to the same conclusion so makes total sense. Thanks!

I'll make the changes and then I think it should be good to go